### PR TITLE
Prevent Test Area photos from showing on the front page #835

### PR DESCRIPTION
**Pull request by hawkishpolicy**
---
name: Pull request
about: Create a pull request
title: ''
labels: ''
assignees: ''

---

## What type of PR is this?(check all applicable)
- [ ] refactor
- [ ] feature
- [ ] bug fix
- [ ] documentation
- [ ] optimization
- [ ] other

### Related Issues

Issue #

### What this PR achieves

Excludes media with Test Area ID from being displayed on the front page. Also assigns Test Area ID to a configurable delimited list inside .env

I also had to update Node to 18.

### CODE REVIEW

File: src/pages/index.tsx [hunk 1]
```diff
@@ -201,7 +202,11 @@ export const getStaticProps: GetStaticProps = async ({ params }) => {
 
   const recentTagsByUsers = await getMediaForFeed(20, 3)
 
-  const recentTags = recentTagsByUsers.flatMap(entry => entry.mediaWithTags)
+  const areaTags = recentTagsByUsers.flatMap(entry => entry.mediaWithTags)
+  const recentTags = areaTags.filter(tag => {
+    return !tag.entityTags.some(entityTag =>
+      testAreaIds.has(entityTag.targetId))
+  })
```
Comment (vnugent) [inline src/pages/index.tsx]: This filter will correctly exclude media with the exact match of NEXT_PUBLIC_TEST_AREA_IDS. 
There's another scenario where we have media tagged in child areas or climbs under 'NEXT_PUBLIC_TEST_AREA_IDS'.   You can use regular expression to match `testAreaIds` and `entityTag.ancestors`.

Example:
A test area (and its children) we want to exclude (NZ test area)
- https://stg.openbeta.io/crag/f72326ab-154b-527b-8bf6-4f9b73e7467c

A photo tagged under `NZ test area / NZ Sport & Trad`
- https://stg.openbeta.io/climbs/2d63fb47-b2eb-42f1-af9f-26d4b7a1ee5a

**Review by vnugent — CHANGES_REQUESTED** [review]

File: src/pages/index.tsx (hunk 1 above)
Comment (hawkishpolicy) [reply src/pages/index.tsx]: You'll have to forgive my ignorance as I am quite new to programming, but could you provide an example of this regex?

I do not completely understand what you are asking.

File: src/pages/index.tsx [hunk 2]
```diff
@@ -201,8 +201,18 @@ export const getStaticProps: GetStaticProps = async ({ params }) => {
...
+  const mediaWithTags = recentTagsByUsers.flatMap(entry => entry.mediaWithTags)
+
+  const recentTags = mediaWithTags.filter(tag => {
+    return tag.entityTags.some(entityTag => {
+      return testAreaIds.every(testId => {
+        const regex = new RegExp(testId, 'g')
+        return !regex.test(entityTag.ancestors)
+      })
+    })
+  })
```
Comment (vnugent) [inline src/pages/index.tsx:215]: The original issue should be expanded to something like "Prevent Test Area  **and their child areas'** photos from showing on the front page"

Consider this scenario:
1. I tagged a climb under NZ Sport & Trad
2. I want to exclude photos in NZ test area as well as those in child areas
3. I specify NZ test area id in .env

Home / New Zealand / **NZ test area** / NZ Sport & Trad

---

If you inspect each entityTag object, you'll see the `ancestors` field containing a list of ancestor IDs which corresponds to the breadcrumbs: New Zealand / **NZ test area** / NZ Sport & Trad

ancestors: '2d2e7df8-8e0c-5cdf-ab35-42a39b964e5a,**f72326ab-154b-527b-8bf6-4f9b73e7467c**,75be7758-98d9-5dd3-a1f5-6c6590b2d5aa'


```
{
  __typename: 'MediaWithTags',
  id: '64c143eada6d989cb734a886',
  username: 'viet-test-profile112aaabdsbdsb',
  mediaUrl: '/u/b9f8ab3b-e6e5-4467-9adb-65d91c7ebe7c/mQw9tHQtnn.jpeg',
  width: 2000,
  height: 2667,
  uploadTime: 1690387434598,
  entityTags: [
    {
      __typename: 'EntityTag',
      id: '64c143eada6d989cb734a885',
      targetId: '2d63fb47-b2eb-42f1-af9f-26d4b7a1ee5a',
      climbName: 'climb one',
      areaName: 'NZ Sport & Trad',
      ancestors: '2d2e7df8-8e0c-5cdf-ab35-42a39b964e5a,f72326ab-154b-527b-8bf6-4f9b73e7467c,75be7758-98d9-5dd3-a1f5-6c6590b2d5aa',
      type: 0
    }
  ]
}
```

**Review by vnugent — APPROVED** [review]

**Comment by vnugent** [comment]
@hawkishpolicy I made a small update to your PR.  We're good to merge now.

**Pull request merged**